### PR TITLE
ENH #15972 added margins_name parameter for crosstab

### LINKED ISSUE #15972: Crosstab support for setting the margins name
`pandas.pivot_table` already provides the parameter `margins_name ` . `pandas.crosstab` currently does not.

### PR DESCRIPTION
**Pull request by cmohl2013**
-  closes #15972
 -  1 tests added / passed
 -  passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 -  whatsnew entry

### CODE REVIEW

File: pandas/core/reshape/pivot.py [hunk 1]
```diff
@@ -411,6 +412,12 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
         If passed, must match number of column arrays passed
     margins : boolean, default False
         Add row/column margins (subtotals)
+    margins_name : string, default 'All'
+        Name of the row / column that will contain the totals
+        when margins is True.
+
+        .. versionadded:: 0.20.0
```
Comment (jreback) [inline pandas/core/reshape/pivot.py]: 0.21.0

File: pandas/tests/reshape/test_pivot.py [hunk 2]
```diff
@@ -1071,6 +1071,42 @@ def test_crosstab_margins(self):
         exp_rows = exp_rows.fillna(0).astype(np.int64)
         tm.assert_series_equal(all_rows, exp_rows)
 
+    def test_crosstab_margins_set_margin_name(self):
+        a = np.random.randint(0, 7, size=100)
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: add the issue number as a comment

File: pandas/tests/reshape/test_pivot.py [hunk 3]
```diff
@@ -1071,6 +1071,42 @@ def test_crosstab_margins(self):
...
+
+        result = crosstab(a, [b, c], rownames=['a'], colnames=('b', 'c'),
+                          margins=True, margins_name='TOTAL')
+
+        assert result.index.names == ('a',)
+        assert result.columns.names == ['b', 'c']
+
+        all_cols = result['TOTAL', '']
+        exp_cols = df.groupby(['a']).size().astype('i8')
+        # to keep index.name
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: blank line here

**Review by jreback — APPROVED** [review]
minor comments. ping on green.

File: pandas/core/reshape/pivot.py [hunk 4]
```diff
@@ -561,8 +571,10 @@ def _normalize(table, normalize, margins):
         elif normalize == "all" or normalize is True:
             column_margin = column_margin / column_margin.sum()
             index_margin = index_margin / index_margin.sum()
-            index_margin.loc['All'] = 1
+            index_margin.loc[margins_name] = 1
             table = concat([table, column_margin], axis=1)
+            print('index margin')
```
Comment (jreback) [inline pandas/core/reshape/pivot.py]: remove prints

File: doc/source/whatsnew/v0.20.0.txt [hunk 5]
```diff
@@ -514,7 +514,7 @@ Other Enhancements
 - ``parallel_coordinates()`` has gained a ``sort_labels`` keyword argument that sorts class labels and the colors assigned to them (:issue:`15908`)
 - Options added to allow one to turn on/off using ``bottleneck`` and ``numexpr``, see :ref:`here <basics.accelerate>` (:issue:`16157`)
 - ``DataFrame.style.bar()`` now accepts two more options to further customize the bar chart. Bar alignment is set with ``align='left'|'mid'|'zero'``, the default is "left", which is backward compatible; You can now pass a list of ``color=[color_negative, color_positive]``. (:issue:`14757`)
-
+- ``Crosstab`` has gained a ``margins_name`` parameter to define the name of the row / column that will contain the totals when margins is True(:issue:`15972`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.0.txt]: move to 0.21.0.txt

**Comment by cmohl2013** [comment]
ok

**Comment by jreback** [comment]
I think you have a conflict on 0.21.0.txt (pls rebase again).

File: doc/source/whatsnew/v0.21.0.txt [hunk 6]
```diff
@@ -35,6 +35,7 @@ Other Enhancements
 - ``RangeIndex.append`` now returns a ``RangeIndex`` object when possible (:issue:`16212`)
 - :func:`to_pickle` has gained a protocol parameter (:issue:`16252`). By default, this parameter is set to `HIGHEST_PROTOCOL <https://docs.python.org/3/library/pickle.html#data-stream-format>`__
 - :func:`api.types.infer_dtype` now infers decimals. (:issue: `15690`)
+- ``Crosstab`` has gained a ``margins_name`` parameter to define the name of the row / column that will contain the totals when margins is True(:issue:`15972`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.21.0.txt]: Could you add a space before the `(` when you rebase.

**Review by TomAugspurger — APPROVED** [review]

File: doc/source/whatsnew/v0.21.0.txt [hunk 7]
```diff
@@ -35,6 +35,7 @@ Other Enhancements
 - ``RangeIndex.append`` now returns a ``RangeIndex`` object when possible (:issue:`16212`)
 - :func:`to_pickle` has gained a protocol parameter (:issue:`16252`). By default, this parameter is set to `HIGHEST_PROTOCOL <https://docs.python.org/3/library/pickle.html#data-stream-format>`__
 - :func:`api.types.infer_dtype` now infers decimals. (:issue: `15690`)
+- ``Crosstab`` has gained a ``margins_name`` parameter to define the name of the row / column that will contain the totals when margins is True. (:issue:`15972`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: :func:`crosstab`

when ``margins=True``

**Review by jreback — APPROVED** [review]
minor comment

**Comment by cmohl2013** [comment]
@jreback 
@TomAugspurger 

Thank you for your checking and helpful comments.

**Comment by TomAugspurger** [comment]
Thanks @cmohl2013 !

**Pull request merged**